presentation-compiler/src/main/dotty/tools/pc/SignatureHelpProvider.scala

```diff
@@ -1,79 +1,134 @@
 package dotty.tools.pc
 
-import scala.jdk.CollectionConverters._
-import scala.meta.pc.OffsetParams
-import scala.meta.pc.SymbolDocumentation
-import scala.meta.pc.SymbolSearch
-
-import dotty.tools.dotc.ast.Trees.AppliedTypeTree
-import dotty.tools.dotc.ast.Trees.TypeApply
-import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.ast.tpd.*
 import dotty.tools.dotc.core.Contexts.*
 import dotty.tools.dotc.core.Flags
 import dotty.tools.dotc.core.Symbols.*
 import dotty.tools.dotc.interactive.Interactive
 import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.parsing.Tokens.closingRegionTokens
+import dotty.tools.dotc.reporting.ErrorMessageID
+import dotty.tools.dotc.reporting.ExpectedTokenButFound
 import dotty.tools.dotc.util.Signatures
-import dotty.tools.dotc.util.Signatures.Signature
 import dotty.tools.dotc.util.SourceFile
-import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.dotc.util.Spans
+import dotty.tools.dotc.util.Spans.Span
 import dotty.tools.pc.utils.MtagsEnrichments.*
-
 import org.eclipse.lsp4j as l
 
+import scala.jdk.CollectionConverters.*
+import scala.jdk.OptionConverters.*
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.SymbolDocumentation
+import scala.meta.pc.SymbolSearch
+
 object SignatureHelpProvider:
 
   def signatureHelp(
       driver: InteractiveDriver,
       params: OffsetParams,
       search: SymbolSearch
   ) =
-    val uri = params.uri()
-    val sourceFile = SourceFile.virtual(params.uri().nn, params.text().nn)
+    val uri = params.uri().nn
+    val text = params.text().nn
+    val sourceFile = SourceFile.virtual(uri, text)
     driver.run(uri.nn, sourceFile)
 
-    given ctx: Context = driver.currentCtx
-
-    val pos = driver.sourcePosition(params)
-    val trees = driver.openedTrees(uri.nn)
-
-    val path = Interactive.pathTo(trees, pos)
+    driver.compilationUnits.get(uri) match
```

Comment:
Why this change?

Reply:
We need untpd trees, and without this you don't have this information in context.